geonode/assets/utils.py

```diff
@@ -98,27 +127,32 @@ def create_asset_and_link(
     clone_files: bool = True,
 ) -> tuple[Asset, Link]:
 
-    asset_handler = handler or asset_handler_registry.get_default_handler()
     asset = link = None
     try:
-        default_title, default_ext = os.path.splitext(next(f for f in files)) if len(files) else (None, None)
+        first_file = next(iter(files)) if len(files) else None
+        filename = (
+            first_file
+            if isinstance(first_file, (str, Path))
+            else getattr(first_file, "name", None) if first_file else None
+        )
+        default_title, default_ext = os.path.splitext(filename) if filename else ("Unknown", None)
         if default_ext:
             default_ext = default_ext.lstrip(".")
         link_type = link_type or find_type(default_ext) if default_ext else None
-
-        asset = asset_handler.create(
-            title=title or default_title or "Unknown",
-            description=description or asset_type or "Unknown",
-            type=asset_type or "Unknown",
+        asset = create_asset(
             owner=owner,
             files=files,
+            handler=handler,
+            title=title or default_title,
+            description=description,
+            asset_type=asset_type,
             clone_files=clone_files,
         )
 
         link = create_link(
             resource,
             asset,
-            asset_handler=asset_handler,
+            asset_handler=handler,
```

Comment:
you removed line 101 
```
asset_handler = handler or asset_handler_registry.get_default_handler()
``` 
when you refactored out the `create_asset' method, so here handler may be null!